sdks/unreal/Agones/Source/Agones/Public/AgonesComponent.h

```diff
@@ -0,0 +1,279 @@
+#pragma once
+
+#include "Classes.h"
+#include "Components/ActorComponent.h"
+#include "CoreMinimal.h"
+#include "Interfaces/IHttpRequest.h"
+
+#include "AgonesComponent.generated.h"
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FAgonesErrorDelegate, const FAgonesError&, Error);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FAllocateDelegate, const FEmptyResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FGameServerDelegate, const FGameServerResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FGetConnectedPlayersDelegate, const FConnectedPlayersResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FGetPlayerCapacityDelegate, const FCountResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FGetPlayerCountDelegate, const FCountResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FHealthDelegate, const FEmptyResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FIsPlayerConnectedDelegate, const FConnectedResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FPlayerConnectDelegate, const FConnectedResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FPlayerDisconnectDelegate, const FDisconnectResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FReadyDelegate, const FEmptyResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FReserveDelegate, const FEmptyResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FSetAnnotationDelegate, const FEmptyResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FSetLabelDelegate, const FEmptyResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FSetPlayerCapacityDelegate, const FEmptyResponse&, Response);
+
+DECLARE_DYNAMIC_DELEGATE_OneParam(FShutdownDelegate, const FEmptyResponse&, Response);
+
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FConnectedDelegate, const FGameServerResponse&, Response);
+
+class FHttpVerb
+{
+public:
+	enum EVerb
+	{
+		Get,
+		Post,
+		Put
+	};
+
+	// ReSharper disable once CppNonExplicitConvertingConstructor
+	FHttpVerb(const EVerb Verb) : Verb(Verb)
+	{
+	}
+
+	FString ToString() const
+	{
+		switch (Verb)
+		{
+			case Post:
+				return TEXT("POST");
+			case Put:
+				return TEXT("PUT");
+			case Get:
+			default:
+				return TEXT("GET");
+		}
+	}
+
+private:
+	const EVerb Verb;
+};
+
+/**
+ * \brief UAgonesComponent is the Unreal Component to call to the Agones SDK.
+ * See - https://agones.dev/ for more information.
+ */
+UCLASS(ClassGroup = (Custom), meta = (BlueprintSpawnableComponent))
+class AGONES_API UAgonesComponent final : public UActorComponent

```

Comment:
Previously Agones plugin settings could be configured in a config file (by having `UCLASS(config = Game, defaultconfig)` as class properties), is that still possible or do these settings need to be set at runtime now?

Reply:
It is possible to do as mentioned we could add the UCLASS config section ... would require some more thinking of where we would want the config for the plugin to live. Can add it onto my back log to investigate.

Reply:
I think that would be good, also to ease the migration from the old plugin to this one. It also makes sense for things like auto connect and health checks to live in a config file as they are picked up immediately once the Agones plugin is created, you'd have to be careful changing these values at runtime at the right time

Reply:
have added port, rate of health check and disableConnect to the config ... this is also added to the docs

Reply:
Nice, can you add `defaultconfig` as well (from https://github.com/googleforgames/agones/pull/1352)?

Reply:
mind blown, didnt even know this existed 😂

Reply:
> mind blown, didnt even know this existed

:sparkles::sparkles::sparkles:  THE POWER OF OPEN SOURCE AT WORK :sparkles::sparkles::sparkles: